### PR TITLE
MM-T346 Public link related to a deleted post should no longer open the file

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Summary

This PR includes an end to end test for verifying the deleted post attachment are indeed deleted when we try to visit their public links

#### Ticket Link
MM-T346

#### Related Pull Requests
none

#### Screenshots
none

### CODE REVIEW

**Comment by M-ZubairAhmed** [comment]
@josephbaylon Hi, thanks for bringing this up, unfortunately my computer doesnt handle running more than one test simulataneously. Also i see that the problem seems to be in beforeEach. If you can let me know i will gladly be able to fix

**Comment by josephbaylon** [comment]
@M-ZubairAhmed I tried locally and just needs `cy.apiAdminLogin();` before `cy.apiInitSetup()..` to reset to admin

**Comment by M-ZubairAhmed** [comment]
@josephbaylon build is failing due to some node error. Can you please check, all other are working fine, is there something i should do from here?

File: e2e/cypress/integration/files_and_attachments/upload_files_spec.js [hunk 1]
```diff
@@ -202,6 +205,79 @@ describe('Upload Files', () => {
...
+                // # Confirm click on the delete button for the post
+                cy.findByText('Delete').should('be.visible').click();
+            });
+
+        // # Try to fetch the url of the attachment we previously deleted
+        cy.get('@publicLinkOfAttachment').then((publicLinkOfAttachment) => {
+            cy.request({url: publicLinkOfAttachment, failOnStatusCode: false}).then((response) => {
+                // * Verify that the link no longer exists in the system
+                expect(response.status).to.be.equal(404);
+            });
```
Comment (saturninoabril) [inline e2e/cypress/integration/files_and_attachments/upload_files_spec.js:278]: I think this is a bonus but it's missing the UI verification on "You should see an error displayed "Error, Unable to get the file info, Back to Mattermost".

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks @M-ZubairAhmed. LGTM, except for one request.

File: e2e/cypress/integration/files_and_attachments/upload_files_spec.js [hunk 2]
```diff
@@ -202,6 +205,83 @@ describe('Upload Files', () => {
             cy.wait(TIMEOUTS.HALF_SEC);
         });
     });
+
+    it.only('MM-T346 Public link related to a deleted post should no longer open the file', () => {
```
Comment (saturninoabril) [inline e2e/cypress/integration/files_and_attachments/upload_files_spec.js:209]:
```suggestion
    it('MM-T346 Public link related to a deleted post should no longer open the file', () => {
```

File: e2e/cypress/integration/files_and_attachments/upload_files_spec.js [hunk 3]
```diff
@@ -202,6 +205,83 @@ describe('Upload Files', () => {
...
+
+        // # Try to fetch the url of the attachment we previously deleted
+        cy.get('@publicLinkOfAttachment').then((publicLinkOfAttachment) => {
+            cy.request({url: publicLinkOfAttachment, failOnStatusCode: false}).then((response) => {
+                // * Verify that the link no longer exists in the system
+                expect(response.status).to.be.equal(404);
+
+                // * Verify that it contains text stating unable to get the file info
+                expect(response.body).to.contain('Unable+to+get+the+file+info.');
+            });
```
Comment (saturninoabril) [inline e2e/cypress/integration/files_and_attachments/upload_files_spec.js:282]: Thanks for updating. Could you add action/check below and see if it works for you?
```javascript
// # Visit the public link
cy.visit(publicLinkOfAttachment, {failOnStatusCode: false});

// * Page should display an error
cy.findByText('Error').should('be.visible');
cy.findByText('Unable to get the file info.').should('be.visible');
cy.findByText('Back to Mattermost').should('be.visible').parent().should('have.attr', 'href', '/');
```

**Review by saturninoabril — CHANGES_REQUESTED** [review]

**Review by saturninoabril — APPROVED** [review]
Thanks @M-ZubairAhmed, LGTM!

**Pull request merged**